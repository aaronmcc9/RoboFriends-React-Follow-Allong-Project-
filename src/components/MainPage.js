import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from "../components/Scroll";
import { ErrorBoundary } from "../components/ErrorBoundary";
import Header from "../components/Header";
import './MainPage.css'

function MainPage(props) {

    const { searchField, onSearchChanged, robots, isPending } = props;

    useEffect(() => {
        props.onRequestRobots();
    }, []);

    const filteredRobots = () => {
        return robots.filter((robot) => {
            return robot.name.toLowerCase().includes(searchField.toLowerCase());
        })
    }

    return isPending ?
        <h1>Loading</h1> :
        (
            <div className="tc">
                <Header />
                <SearchBox searchChange={onSearchChanged} />
                <Scroll>
                    <ErrorBoundary>
                        <CardList robots={filteredRobots()} />
                    </ErrorBoundary>
                </Scroll>
            </div>
        );
}

export default MainPage;