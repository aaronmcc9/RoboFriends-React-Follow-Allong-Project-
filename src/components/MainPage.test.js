import { shallow, mount, render } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

//before each test this should run
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    }

    wrapper = shallow(<MainPage {...mockProps} />)
})

it('expects to render MainPage component without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})

it('filters the robots correctly', () => {

    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            username: 'JohnJacobs'
        }],
        searchField: 'John'
    }

    const wrapper = shallow(<MainPage {...mockProps2} />)
    expect(wrapper.instance().filteredRobots()).toEqual([{
        id: 3,
        name: 'John',
        username: "JohnJacobs",
        isPending: false
    }]);
})