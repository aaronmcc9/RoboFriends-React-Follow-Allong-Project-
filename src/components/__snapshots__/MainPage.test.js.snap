// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expects to render MainPage component without crashing 1`] = `ShallowWrapper {}`;
